tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -438,23 +439,33 @@ function test_subdomain_switching_to_www_does_not_cause_sync() {
 		$this->assertEquals( $original_site_url, $this->server_replica_storage->get_callable( 'site_url' ) );
 	}
 
-	function test_only_syncs_if_is_admin_and_not_cron() {
-		// non-admin
-		set_current_screen( 'front' );
-		$this->sender->do_sync();
-		$this->assertEquals( null, $this->server_replica_storage->get_callable( 'site_url' ) );
+	function test_sync_limited_set_of_callables_if_cron() {
+		$all_callables = array_keys( Defaults::get_callable_whitelist() );
+		$always_updated = Callables::ALWAYS_SEND_UPDATES_TO_THESE_OPTIONS;
 
-		set_current_screen( 'post-user' );
+		foreach ( $always_updated as $option ) {
+			if ( array_key_exists( $option, Callables::OPTION_NAMES_TO_CALLABLE_NAMES ) ) {
+				unset( $always_updated[ $option ] );
```

Comment:
This seems wrong - it doesn't seem like we maintain array keys for `Callables::ALWAYS_SEND_UPDATES_TO_THESE_OPTIONS`. Should we use `array_search` to locate the right key instead?

Reply:
@tyxla , `array_key_exists` is the correct function here, because we want to see if the array/map `Callables::OPTION_NAMES_TO_CALLABLE_NAMES` has a key `$option`. If it does, that means the callable name differs from the option name, and we replace the option name with the callable name in the list of callables to always update.

Reply:
I think we're talking about different things. 

`$always_updated` uses `Callables::OPTION_NAMES_TO_CALLABLE_NAMES`, and the latter is an array with numeric keys, as defined in the class. So basically, how would we have `$option` as a key there? And then we're trying to unset that key?